capstone/src/test/java/com/google/sps/servlets/CheckNewUserServletTest.java

```diff
@@ -0,0 +1,136 @@
+package com.google.sps.servlets;
```

Comment:
Sorry I can't comment on a file you're not editing in the pull request. This also may be fully out of the scope of this pull request, but I have a question. Looking at CheckNewUserServlet:

```java
    boolean isUserNew = false;

    if (userService.isUserLoggedIn()) {
      String userId = userService.getCurrentUser().getUserId();

      // Check if user already exists in Datastore. If so, do nothing.
      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
      Key entityKey = KeyFactory.createKey("User", userId);
      try {
        datastore.get(entityKey);
      } catch (EntityNotFoundException e) {
        // If the user doesn't exist in Datastore.
        isUserNew = true;
      }
    }

    String json = String.format("{\"isUserNew\": %b}", isUserNew);
```

If `userService.isUserLoggedIn()` is false, you return a normal response where you claim `{isUserNew: false}` but you would have no way of knowing since `isUserLoggedIn()` returned false, right? I may be missing something, but should you error out if the userService can't identify the user? Identify them as a new user? Redirect them to a different page?

Reply:
That's true, I think it should send an error. Because ideally /checknewuserservlet is only called after /loginservlet returns true, so the user should never be at this point and logged out.

Also I noticed you mentioned creating an abstract class AuthenticatedServlet that verifies authentication in Lucy's PR - if we did that, I guess I wouldn't need the additional check at all?